src/sentry/workflow_engine/endpoints/validators.py

```diff
@@ -0,0 +1,148 @@
+from django.db import router, transaction
+from rest_framework import serializers
+from rest_framework.fields import Field
+
+from sentry import audit_log
+from sentry.api.serializers.rest_framework import CamelSnakeSerializer
+from sentry.issues import grouptype
+from sentry.issues.grouptype import GroupType
+from sentry.utils.audit import create_audit_entry
+from sentry.workflow_engine.models import DataConditionGroup, Detector
+from sentry.workflow_engine.models.data_condition import Condition, DataCondition
+from sentry.workflow_engine.types import DetectorPriorityLevel
+
+
+class BaseDataConditionValidator(CamelSnakeSerializer):
```

Comment:
I feel like we're going to have a _lot_ of condition validators based on these examples, how would you feel about changing `validators` to be a directory, then have the `__init__` file export these in a unified way?

I think it might be a little easier to grok "this is where the condition is" if we split it out to different validators (detector, data_source, data_condition etc)

Reply:
That might make sense, but we should potentially do it as needed. One thing to remember is that `workflow_engine` is the place for generic conditions/validators. Things for issue rule processing should probably be in the issues/ folder